website/pages/en/cookbook/subgraph-uncrashable.mdx

```diff
@@ -4,7 +4,7 @@ title: Safe Subgraph Code Generator
 
 [Subgraph Uncrashable](https://float-capital.github.io/float-subgraph-uncrashable/) is a code generation tool that generates a set of helper functions from the graphql schema of a project. It ensures that all interactions with entities in your subgraph are completely safe and consistent.
 
-### Why integrate with Subgraph Uncrashable?
+## Why integrate with Subgraph Uncrashable?
```

Comment:
Was this rewritten with `pnpm lint:fix`, or did you change it manually? If the former, what rule is responsible for it? I'm not sure we always want to follow h2 => h3 => h4.

Reply:
I changed it manually, remark lint doesn’t support auto fix

Reply:
> I'm not sure we always want to follow h2 => h3 => h4.

it's a good practice, especially for TOC, so every heading will have right amount of `margin-left`

<img width="230" alt="image" src="https://user-images.githubusercontent.com/7361780/228081595-2357e24b-0df8-4f58-b2e1-bcd3720c29bd.png">

Reply:
also there is a rule for this case that we could also extend https://github.com/remarkjs/remark-lint/tree/main/packages/remark-lint-heading-increment

Reply:
Hmm, so I would tend to be a bit less strict here and only keep the custom rule that you created (`remark-lint-restrict-elements` to forbid `h1` altogether). The `remark-lint-first-heading-level` rule is a bit strange: it will error if the first heading is an `h3` instead of an `h2`, but it won't care at all if an `h2` is immediately followed by an `h5` instead of an `h3`. I guess that's what `remark-lint-heading-increment` is for, but I feel like we should allow for exceptions and leave it up to code reviewers. For instance, for FAQs, I find that smaller questions look better (and I'm not bothered by the subtle margin difference in the TOC that you mentioned):

_Before_

<img width="1346" alt="CleanShot 2023-03-27 at 22 31 17@2x" src="https://user-images.githubusercontent.com/1059139/228112142-5cf023b4-4cc7-4169-9720-3eec82f0a26a.png">

_After_

<img width="1346" alt="CleanShot 2023-03-27 at 22 31 25@2x" src="https://user-images.githubusercontent.com/1059139/228112169-8fc18664-f497-49f8-b633-f4d45c1ee956.png">

I would like someone else to chime on this though. I could easily be convinced.

Reply:
>  I guess that's what remark-lint-heading-increment is for,

exactly

> I would like someone else to chime on this though. I could easily be convinced.

sure, let's hear through others and anyway I could do a rollback of changes

Reply:
Cool, tagging @MichaelMacaulay since he contributes the most to actual content, and @Web3Slimchance, @azf20, @nenadjaja, and @schmidsi because they might have an opinion. The question is: do we want to allow skipping heading levels (other than `h1` which is always there at the top), e.g.  having `h3`s without an `h2`, like in the first FAQs screenshot I posted above?

Reply:
@B2o5T – Michael and Adam are OK with `remark-lint-first-heading-level` + `remark-lint-heading-increment`. Let's keep them and merge this when you can!